src/Cli/dotnet/Installer/Windows/Security/AuthentiCode.cs

```diff
@@ -1,190 +1,61 @@
-            {
-                cbStruct = (uint)Marshal.SizeOf(typeof(WinTrustFileInfo)),
-                pcwszFilePath = Path.GetFullPath(path),
-                hFile = IntPtr.Zero,
-                pgKnownSubject = IntPtr.Zero
-            };
-
-            WinTrustData data = new()
-            {
-                cbStruct = (uint)Marshal.SizeOf(typeof(WinTrustData)),
-                dwProvFlags = 0,
-                dwStateAction = StateAction.WTD_STATEACTION_IGNORE,
-                dwUIChoice = UIChoice.WTD_UI_NONE,
-                dwUIContext = 0,
-                dwUnionChoice = UnionChoice.WTD_CHOICE_FILE,
-                fdwRevocationChecks = RevocationChecks.WTD_REVOKE_NONE,
-                hWVTStateData = IntPtr.Zero,
-                pWinTrustInfo = Marshal.AllocHGlobal(Marshal.SizeOf(typeof(WinTrustFileInfo))),
-                pPolicyCallbackData = IntPtr.Zero,
-                pSIPClientData = IntPtr.Zero,
-                pwszURLReference = IntPtr.Zero
-            };
-
-            Marshal.StructureToPtr(fileInfo, data.pWinTrustInfo, false);
-
-            IntPtr pGuid = Marshal.AllocHGlobal(Marshal.SizeOf(typeof(Guid)));
-            IntPtr pData = Marshal.AllocHGlobal(Marshal.SizeOf(typeof(WinTrustData)));
-            Marshal.StructureToPtr(data, pData, true);
-            Marshal.StructureToPtr(NativeMethods.WINTRUST_ACTION_GENERIC_VERIFY_V2, pGuid, true);
-
-            uint result = NativeMethods.WinVerifyTrust(IntPtr.Zero, pGuid, pData);
-
-            Marshal.FreeHGlobal(pGuid);
-            Marshal.FreeHGlobal(pData);
-            Marshal.FreeHGlobal(data.pWinTrustInfo);
-
-            // The return value is a LONG, not an HRESULT.
-            // Do not use HRESULT macros such as SUCCEEDED to determine whether the function succeeded.
-            // Instead, check the return value for equality to zero.
-            return result == 0;
-        }
-
-        /// <summary>
-        /// Determines if the file contains a signature associated with one of the specified organizations. The primary certificate is queried
-        /// before looking at nested certificates.
-        /// </summary>
-        /// <param name="path">The path of the signed file.</param>
-        /// <param name="organizations">A set of trusted organizations.</param>
-        /// <returns><see langword="true"/> if organization described in the certificate subject matches any of the specified 
-        /// organizations; <see langword="false"/> otherwise.</returns>
-        internal static bool IsSignedByTrustedOrganization(string path, params string[] organizations)
+        /// <param name="path">The path of the file to verify.</param>
+        /// <param name="cacheOnlyRevocationChecks">Prevents revocation checks over the network when set to <see langword="true"/>.</param>
+        /// <returns>0 if successful.</returns>
+        public static unsafe int IsSigned(string path, bool cacheOnlyRevocationChecks = false)
```

Comment:
`cacheOnlyRevocationChecks` is poorly named, as it implies that only revocation checks will be cached when `true`.  (Windows may cache other things, like intermediate certificate authority certificates.)  How about `allowOnlineRevocationChecks`?

Reply:
Agree, the suggested name is more clear about the intent.

Reply:
Renamed the policy key to `AllowOnlineRevocationChecks`